### ISSUE TITLE
Windows build fails

### ISSUE THREAD

**Issue by isuruf** [Normaliz/Normaliz#282]
```
lld-link: error: undefined symbol: int volatile nauty_kill_request
```

**Comment by w-bruns** [comment]
This is a global variable defined by nauty in the version 27rc2.  I am not sure about older versions and  I am not sure how it is defined, but grep shows

nautil.c:volatile int nauty_kill_request = 0;   /* no TLS_ATTR on purpose */

**Comment by w-bruns** [comment]
Could you repair it?

**Comment by isuruf** [comment]
I'm not sure how. Any ideas?

cc @saraedum, @videlec

**Comment by saraedum** [comment]
@isuruf: Could you explain the context? Is this on conda-forge?

**Comment by isuruf** [comment]
@saraedum, yes. https://github.com/conda-forge/normaliz-feedstock/pull/9

**Comment by saraedum** [comment]
The symbol also exists in 2.6:
```
  *       18-Jan-12 : changes for version 2.6 :                                  *
  *                 - declare nauty_kill_request       
```

**Comment by saraedum** [comment]
Unfortunately, nauty's build log on azure is gone. Maybe we can just upgrade nauty. Chances are they fixed something in there.

**Comment by saraedum** [comment]
@isuruf: How does the nauty Windows build work anyway? I cannot see a `source` section in the recipe.

**Comment by saraedum** [comment]
Upgrade is here: https://github.com/conda-forge/nauty-feedstock/pull/7

**Comment by saraedum** [comment]
@w-bruns fixed, see the patch https://github.com/conda-forge/normaliz-feedstock/pull/9. You might want to upstream this change into nauty.

**Comment by w-bruns** [comment]
It would be acceptable to omit the use of nauty_kill_request under MS-Windows or even in general because the interruption of nauty does not work well with the version I link into Normaliz. However, I could test a preliminary version of the next release. Then it will be much better.

**Comment by saraedum** [comment]
Sure, if you don't use it anymore that's also fine. But the patch is quite small so you might just want to add this to normaliz. I notified upstream about this btw.

**Comment by w-bruns** [comment]
Where is the patch?

**Comment by saraedum** [comment]
In the changeset of the linked PR.

**Comment by isuruf** [comment]
https://github.com/conda-forge/normaliz-feedstock/blob/298e0678a3b8f47bbbc51650e1fca8e74ac4c127/recipe/extern-kill.patch

**Comment by w-bruns** [comment]
Before  I proceed, two questions.

1) Is the problem you encountered with the Windows build
a weak point in the Normaliz code that some compilers generously ignore or a problem arising from the nauty installation under Windiws?

I am asking for curiosity.

2. More importantly, do you think I should add the patch somehow to Normaliz 3.8.1 or use it for Normaliz 3.8.2?

It is clear that in the long run we want to be able to interrupt nauty calculations within Normaliz.

**Comment by saraedum** [comment]
> Before I proceed, two questions.
> 
>     1. Is the problem you encountered with the Windows build
>        a weak point in the Normaliz code that some compilers generously ignore or a problem arising from the nauty installation under Windiws?

It's an error in the nauty headers. C++ compilers mangle names, the ones you see when running `objdump -T` on an `.so` file. Run with `-TC` to see the original names. This mangling needs to be consistent when building a library and when linking to it. The problem here is that the normaliz compilation mistakes nauty's kill switch as a C++-mangled name when it is actually a C symbol. On common compilers for Linux/macOS this does not seem to make a difference but on Windows the mangling seems to be different (probably because of the volatile.)

>     1. More importantly, do you think I should add the patch somehow to Normaliz 3.8.1 or use it for Normaliz 3.8.2?

It only seems to affect Windows builds with some compilers. It's fine to postpone it to 3.8.2 I think.

**Comment by fingolfin** [comment]
I believe this was resolved in f836c224dfeb7663e78656fde5c0421f853d70a7 which is in 3.8.2 resp. 3.8.3
